boards/nordic/.cargo/config.toml

```diff
@@ -0,0 +1,14 @@
+# Licensed under the Apache License, Version 2.0 or the MIT License.
```

Comment:
I'm skittish about having these configs in directories other than the board crate itself. It would seem to me like this inconsistency could result in confusion down the line. Same applies to, e.g., the LiteX boards.

Reply:
Ah... you're probably right. The small bit of savings isn't worth the inconsistency. Will fix.

Reply:
The linker file is shared, what's the difference?

Reply:
I don't feel particularly strongly about this, but one thing that's different from our `build.rs`, linker files, `cargo/*.toml` files is that these are all shared resources that are explicitly referenced by something that's in a board directory.

On the other hand, `.cargo/config.toml` files in _any_ parent directory are implicitly picked up by Cargo. Once consequence of relying on configs in parent directories then is that, were you to copy a Nordic board crate to somewhere else (e.g., use it as a template for a top-level board crate in `boards/`), it would silently no longer take this shared config into account. This is different from, e.g., our linker script, where you'd get an error because the relative path to it has changed.

Reply:
I could get behind a restriction that .cargo dirs can only exist in a folder with a Cargo.toml in Tock.

Reply:
> I could get behind a restriction that .cargo dirs can only exist in a folder with a Cargo.toml in Tock.

I'd support that, with the additional restriction that that `Cargo.toml` must not be a workspace root `Cargo.toml` (i.e., only actual crates should have a `.cargo/config.toml`). The rest we'd solve with includes, which implies that their contents will not be hidden.